### DISCUSSION TITLE
how to compile to .js file not a .mjs file

### DISCUSSION THREAD

**Discussion by Binaryify** [nuxt/nuxt#17037] [Questions] [answered]
Android 8 seem like don't support .mjs file or esm

**Comment by Silbad** [comment]
Hello same problem, can't change server configuration
![image](https://user-images.githubusercontent.com/31576982/183973711-5bac9e87-e0e6-4da4-aec2-db7cff557823.png)
https://github.com/GeoffreyBooth/js-mjs-mime-type-test
Thanks

**Comment by Eliophot** [comment]
Hello same pb on OVH Web Cloud...

**Comment by danielroe** [comment, marked as answer]
I've opened https://github.com/nuxt/framework/pull/6505 to allow this to be done more easily (see PR description).

But for now you can implement like this:

```js
nuxt.hook('vite:extendConfig', (config, { isClient }) {
  if (isClient) {
    // rc6
    config.build.rollupOptions.output.chunkFileNames = '[hash].mjs'
    // rc7
    config.build.rollupOptions.output.chunkFileNames = '_nuxt/[hash].mjs'
  }
})
```

**Reply by sunnywanggit** [reply]
how can i use this? please

**Reply by danielroe** [reply]
You can use the edge channel where .js is the default extension: https://v3.nuxtjs.org/guide/going-further/edge-channel/
